src/providers/oracle/qgsoraclecolumntypetask.cpp

```diff
@@ -0,0 +1,92 @@
+/***************************************************************************
+ qgscolumntypetask.cpp - lookup oracle geometry type and srid in a thread
+                              -------------------
+begin                : 3.1.2012
+copyright            : (C) 2012 by Juergen E. Fischer
+email                : jef at norbit dot de
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsoraclecolumntypetask.h"
+#include "qgslogger.h"
+#include "qgsoracleconnpool.h"
+
+#include <QMetaType>
+
+QgsOracleColumnTypeTask::QgsOracleColumnTypeTask( const QString &name, const QString &limitToSchema, bool useEstimatedMetadata, bool allowGeometrylessTables )
+  : QgsTask( tr( "Scanning tables for %1" ).arg( name ) )
+  , mName( name )
+  , mSchema( limitToSchema )
+  , mUseEstimatedMetadata( useEstimatedMetadata )
+  , mAllowGeometrylessTables( allowGeometrylessTables )
+{
+  qRegisterMetaType<QgsOracleLayerProperty>( "QgsOracleLayerProperty" );
+}
+
+bool QgsOracleColumnTypeTask::run()
+{
+  QString conninfo = QgsOracleConn::toPoolName( QgsOracleConn::connUri( mName ) );
+  QgsOracleConn *conn = QgsOracleConnPool::instance()->acquireConnection( conninfo );
+  if ( !conn )
+  {
+    QgsDebugMsg( "Connection failed - " + conninfo );
+    return false;
+  }
+
+  emit progressMessage( tr( "Retrieving tables of %1…" ).arg( mName ) );
```

Comment:
Would be nice if we could show this message in task manager, but not a blocker.

Reply:
As far as I know, there is no way to do this with current TaskManager API. There is only description task that could be updated but it doesn't look like the appropriate way because we would mix two different type of information.